pkg/hsm/awscloudhsm/cloudhsm.go

```diff
@@ -0,0 +1,150 @@
+package awscloudhsm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	verifyv1beta1 "github.com/alphagov/verify-metadata-controller/pkg/apis/verify/v1beta1"
+	"github.com/alphagov/verify-metadata-controller/pkg/hsm"
+	"github.com/labstack/gommon/log"
+	"gopkg.in/yaml.v2"
+)
+
+var _ hsm.Client = &Client{}
+
+type Client struct{}
+
+func (c *Client) CreateRSAKeyPair(label string, hsmCreds hsm.Credentials) (publicCert []byte, err error) {
+	log.Info("cloudhsmtool",
+		"command", "genrsa",
+		"label", label,
+	)
+	cmd := exec.Command("/cloudhsmtool/build/install/cloudhsmtool/bin/cloudhsmtool",
+		"genrsa", label,
+	)
+	cmd.Stderr = nil // when nil stderr output is captured in err from Output
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("HSM_USER=%s", hsmCreds.User),
+		fmt.Sprintf("HSM_PASSWORD=%s", hsmCreds.Password),
+		fmt.Sprintf("HSM_IP=%s", hsmCreds.IP),
+	)
+	cert, err := cmd.Output()
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate rsa key for %s: %s", label, err)
+	}
+	if !strings.Contains(string(cert), "--BEGIN CERTIFICATE--") {
+		return nil, fmt.Errorf("generated %s certificate does not appear to be a valid PEM format: %s", label, cert)
+	}
+	return cert, nil
```

Comment:
Could an `info` log message be added to indicate successful completion?

Reply:
yep